website/js/index.js

```diff
@@ -475,10 +464,14 @@ const IsoGrid = React.createClass({
                 }
             });
         }
-        else {
+    },
+
+    relayout: function(fullRefresh) {
+        if (fullRefresh) {
             this.masonry.reloadItems();
-            this.masonry.arrange();
         }
+
+        this.masonry.arrange();
```

Comment:
does reloadItems() encompass arrange() or are both necessary in the case of a fullRefresh?

Reply:
It's necessary to call `reloadItems()` when children are added or removed from IsoGrid, which currently occurs when you change modes. `arrange()` just moves the existing children around, which is what we want in most cases when tiles change size.

More in depth, `arrange()` seems to maintain some bookkeeping information between calls for interpolating the positions of elements that `reloadItems()` clears, which is why I made calling it conditional. This shows up as a race condition when you interrupt a layout-in-progress with `reloadItems(); arrange();`-- the tiles end up incorrectly positioned due to, I assume, not being aware of the move in progress.